### PR TITLE
add `metric_stats_to_sync`, `enable_check_large_volume`, `enable_logs_sync`

### PR DESCRIPTION
**Pull request by xp-1000**
and convert `custom_namespace_sync_rule` to a list `custom_namespace_sync_rules` and make it dynamic to allow user defining multiple rules.

### CODE REVIEW

File: cloud/aws/policies.tf [hunk 1]
```diff
@@ -59,10 +59,6 @@ resource "aws_iam_policy" "sfx_policy" {
             "lambda:GetAlias",
             "lambda:ListFunctions",
             "lambda:ListTags",
-            "logs:DeleteSubscriptionFilter",
-            "logs:DescribeLogGroups",
```
Comment (pdecat) [inline cloud/aws/policies.tf:63]: Are you confident these removed permissions are not needed the basic AWS integration?

File: cloud/aws/policies.tf [hunk 2]
```diff
@@ -122,6 +115,32 @@ resource "aws_iam_policy" "sfx_metric_streams_policy" {
...
+            "logs:DeleteSubscriptionFilter",
+            "logs:DescribeLogGroups",
+            "logs:DescribeSubscriptionFilters",
+            "logs:PutSubscriptionFilter",
+            "s3:GetBucketLogging",
+            "s3:GetBucketNotification",
+            "s3:PutBucketNotification"
+        ],
+        "Effect": "Allow",
+        "Resource": "*"
```
Comment (pdecat) [inline cloud/aws/policies.tf:137]: Can't we target specific Cloudwatch logs and S3 resources?

File: cloud/aws/README.md [hunk 3]
```diff
@@ -129,12 +136,22 @@ provider "aws" {
...
+
 - When enabling AWS Cloudwatch Metric Streams with `use_metric_streams_sync = true`, you also need to set `create_metric_streams_iam` to `true`. Then you need to create the Kinesis Data Firehose resources by yourself in each of the regions where you want to collect metrics from. You can use the [`aws-cloudwatch-metric-stream`](https://github.com/claranet/terraform-signalfx-integrations/tree/master/cloud/aws-cloudwatch-metric-stream) module to do this.
 - When disabling AWS Cloudwatch Metric Streams, make sure to apply in two phases:
   - first change the `use_metric_streams_sync` parameter from `true` to `false` and run `terraform apply` to let it deprovision the AWS Cloudwatch Metric Streams resources it created,
   - then change `create_metric_streams_iam` from `true` to `false` and run `terraform apply` to destroy the IAM role.
     If you do not follow that process, the AWS integration will end up in `CANCELATION_FAILED` status.
 
+### CloudWatch Logs sync (BETA)
+
+- When enabling AWS Cloudwatch Logs sync with `enable_logs_sync = true`, you also need to set `create_logs_iam` to `true`.
```
Comment (pdecat) [inline cloud/aws/README.md:150]: Don't you need an additional module to deploy the Splunk AWS Log Collector Lambda function used by this solution in each region, like we did for AWS Cloudwatch Metric Streams with [`aws-cloudwatch-metric-stream`](https://github.com/claranet/terraform-signalfx-integrations/tree/master/cloud/aws-cloudwatch-metric-stream)?

https://docs.splunk.com/Observability/gdi/get-data-in/connect/aws/aws-wizardconfig.html#choose-a-cloudformation-template

 https://o11y-public.s3.amazonaws.com/aws-cloudformation-templates/release/template_logs_regional.yaml

File: cloud/aws/policies.tf (hunk 2 above)
Comment (xp-1000) [reply cloud/aws/policies.tf:137]: I don't think so given that each service can store logs in different buckets (e.g. for ELBs) and splunk does not target anything from their official cloudformation template

File: cloud/aws/README.md (hunk 3 above)
Comment (xp-1000) [reply cloud/aws/README.md:150]: of course we will need a separate module like for metric streams. This PR focus on bringing all features of the integration itself to be future proof. I cannot implement full support for logs for now while we don't have access to log observer yet but this should at least prepare its integration when available.

Comment (xp-1000) [reply cloud/aws/README.md:150]: https://github.com/claranet/terraform-signalfx-integrations/issues/52 created

File: cloud/aws/policies.tf (hunk 1 above)
Comment (xp-1000) [reply cloud/aws/policies.tf:63]: Not really, I rely on https://registry.terraform.io/providers/splunk-terraform/signalfx/latest/docs/resources/aws_integration#enable_logs_sync description but will ask to support

Comment (xp-1000) [reply cloud/aws/policies.tf:63]: Hello @pdecat now we are sure and the official doc has been updated: https://docs.splunk.com/Observability/gdi/get-data-in/connect/aws/aws-apiconfig.html#review-aws-iam-policy-permissions-by-observability-cloud-feature

So the changes was ok according to it and in the current state this modules uses:
- a common policy which is a merge of `Tag and property collection` and `metric collection`
- a policy dedicated to logs including only the required permission in addition to this common policy
- a policy dedicated to metric streams

Everything seem good for except if we prefer to split the common policy between 2 policies: metric collection vs tags sync but it seems useless for me.

**Pull request merged**